test/integration/resourcemanager/managedresource/reconciler_test.go

```diff
@@ -254,6 +309,539 @@ var _ = Describe("ManagedResource controller tests", func() {
+					Class:      pointer.String("test"),
+					SecretRefs: []corev1.LocalObjectReference{{Name: secretForManagedResource.Name}},
+				},
+			}
+		})
+
+		AfterEach(func() {
+			Expect(testClient.Delete(ctx, managedResource)).To(Or(Succeed(), BeNotFoundError()))
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(BeNotFoundError())
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)).To(BeNotFoundError())
+			}, time.Minute, 5*time.Second).Should(Succeed())
+
+			Expect(testClient.Delete(ctx, secretForManagedResource)).To(Or(Succeed(), BeNotFoundError()))
+		})
+
+		It("should not reconcile ManagedResource of any other class except the default class", func() {
+			Expect(testClient.Create(ctx, secretForManagedResource)).To(Succeed())
+			Expect(testClient.Create(ctx, managedResource)).To(Succeed())
+
+			Consistently(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)
+			}, 5*time.Second, time.Second).Should(BeNotFoundError())
+
+			Consistently(func(g Gomega) bool {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(Succeed())
+				condition := gardenerv1beta1helper.GetCondition(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+				return condition == nil
+			}, 10*time.Second, time.Second).Should(BeTrue())
+		})
+	})
+
+	Context("#Reconciliation Modes/Annotations", func() {
+		BeforeEach(func() {
+			configMap = &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: corev1.SchemeGroupVersion.String(),
+					Kind:       "ConfigMap",
+				}, ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: namespaceName,
+				}}
+
+			secretForManagedResource = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: namespaceName,
+				},
+			}
+
+			managedResource = &resourcesv1alpha1.ManagedResource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-managedresource",
+					Namespace: namespaceName,
+				},
+				Spec: resourcesv1alpha1.ManagedResourceSpec{
+					Class:      pointer.String(filter.ResourceClass()),
+					SecretRefs: []corev1.LocalObjectReference{{Name: secretForManagedResource.Name}},
+				},
+			}
+
+			DeferCleanup(func() {
+				Expect(testClient.Delete(ctx, managedResource)).To(Or(Succeed(), BeNotFoundError()))
+				Eventually(func(g Gomega) {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(BeNotFoundError())
+				}, time.Minute, 5*time.Second).Should(Succeed())
+
+				Expect(testClient.Delete(ctx, secretForManagedResource)).To(Or(Succeed(), BeNotFoundError()))
+			})
+		})
+
+		Context("#Ignore Mode", func() {
+			It("should not apply/create resources and remove the resource from the ManagedResource status having ignore mode annotation", func() {
```

Comment:
```suggestion
			It("should not update/re-apply resources having ignore mode annotation and remove them from the ManagedResource status ", func() {
```
This annotation is only used for migration purposes so I think we can phrase it something like this rather than using the word create.